src/test/numa.c

```diff
@@ -39,12 +39,21 @@ int main(void) {
   ret = mbind(p, 16 * page_size, MPOL_PREFERRED, NULL, 0, MPOL_MF_MOVE);
   test_assert(ret == 0 || (ret == -1 && errno == ENOSYS));
 
+  // sanity check
   ret = set_mempolicy(0, NULL, 0);
   test_assert(ret == 0);
-
   ret = get_mempolicy(NULL, NULL, 0, NULL, 0);
   test_assert(ret == 0);
 
+  // test in and out params
+  unsigned long nodemask = 0x1;
+  ret = set_mempolicy(MPOL_BIND, &nodemask, 8 * sizeof(nodemask) + 1);
```

Comment:
To be completely honest, I must have read that man page 5 or 6 times and I still don't fully understand what the syntax is trying to accomplish. I did, however, take a look at the kernel source ( https://github.com/torvalds/linux/blob/master/mm/mempolicy.c#L1322 ). It looks to me like it doesn't matter whether we put 64 or 65 in for `maxnode`.

In the case of 64: the syscall will do `ALIGN(63,64)` which == `64`.

In the case of 65: the syscall will do `ALIGN(64,64)` which == `64`.

The link I gave earlier to the LKML also seems to indicate 65 is the correct value, but that email is also 13 years old.